billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/util/PTFinancialValidator.java

```diff
@@ -35,10 +38,23 @@ public boolean isValid() {
             return false;
         }
 
-        // Fist digit must be 1, 2, 5, 6, 8 or 9
-        char firstDigit = this.financialID.charAt(0);
-        String validChars = "125689";
-        if (validChars.indexOf(firstDigit) == -1) {
+        List<Character> firstDigits = Lists.charactersOf("123568");
+        boolean validFirstDigit = firstDigits
+                .stream()
+                .map(c -> financialID.charAt(0) == c).filter(b -> b)
+                .findAny()
+                .orElse(false);
+
+        List<String> firstDoubleDigits =
+                Lists.newArrayList("45", "70", "71", "72", "74", "75", "77", "79", "90", "91", "98", "99");
+        boolean validDoubleDigits = firstDoubleDigits
+                .stream()
+                .map(c -> financialID.substring(0,  2).equals(c))
+                .filter(b -> b)
+                .findAny()
+                .orElse(false);
+
+        if(!validFirstDigit && !validDoubleDigits){
```

Comment:
This was not easy to read at first for me, if you agree wouldn't it be easier to negate the expression !A && !B => false would be equivalent A || B => true.

Reply:
that is not equivalent. 

With ||, it should be `if(!(validFirstDigit || validDoubleDigits))`. 

that does not read any easier.